source/tutorials/source-file-selection.md

```diff
@@ -0,0 +1,645 @@
+  } ''
+    (
+      set -x
+      cat "$stringFile"
+    )
+  ''
+  ```
+  :::
+
+These all do the same when built:
+```shell-session
+$ nix-build -A package
+building '/nix/store/9fi0khrkmqw5srjzjsfa0b05hf8div4c-file-coercion.drv'...
+++ cat /nix/store/j5lwpnlfrngks3bpidfr5hcrhgq0fy78-string.txt
+This is a string
+```
+
+As you can see, the `string.txt` file was hashed and added to the store,
+which then allowed the build to access it.
+
+The underlying functionality is the same as `nix-store --add` on an absolute path:
+```shell-session
+$ nix-store --add $(pwd)/string.txt
+/nix/store/j5lwpnlfrngks3bpidfr5hcrhgq0fy78-string.txt
+```
+
+### Coercion of directory paths to strings
+
+This path coercion also works on directories the same as it does on files, let's try it out:
+
+```nix
+{ runCommand, tree }:
+runCommand "directory-coercion" {
+  # To nicely show path contents
+  nativeBuildInputs = [ tree ];
+} ''
+  tree ${./.}
+''
+```
+
+Running it gives us:
+```shell-session
+$ nix-build -A package
+building '/nix/store/6ybg4v48xy8azhrnfdccdmhd2gr938f5-directory-interpolation.drv'...
+/nix/store/xdfchqpfx20ar9jil9kys99wc6hnm9zx-select
+|-- default.nix
+|-- nix
+|   |-- sources.json
+|   `-- sources.nix
+|-- package.nix
+`-- string.txt
+```
+
+It's also very common to use this for the [`src`](https://nixos.org/manual/nixpkgs/stable/#var-stdenv-src) variable in `stdenv.mkDerivation` like so:
+
+```nix
+{ stdenv, tree }:
+stdenv.mkDerivation {
+  name = "directory-coercion";
+  src = ./.;
+  nativeBuildInputs = [ tree ];
+  postInstall = ''
+    touch created-in-the-build
+    tree
+  '';
+}
+```
+
+Setting `src` will copy the resulting store path into the build directory and mark it as mutable.
+For the many commands that expect to be able to write to the current directory, this is great:
+
+```shell-session
+$ nix-build -A package
+building '/nix/store/2cqd93fpnb4vqwkwmbl66dbxhndq1vhh-directory-coercion.drv'...
+unpacking sources
+unpacking source archive /nix/store/178fbwa8iwdl6b85yafksdbwlxf6mjca-select
+[...]
+.
+|-- created-in-the-build
+|-- default.nix
+|-- nix
+|   |-- sources.json
+|   `-- sources.nix
+|-- package.nix
+`-- string.txt
+```
+
+However there are some subtle problems with this approach:
+- Note how the name of the store path ends with `-select`.
+  So the name of the local directory influenced the result.
+
+  This means that whenever you rename the project directory
+  or a collegue runs it in a different directory name,
+  you're going to get different build results!
+
+- All files in the directory are unconditionally added to the Nix store.
+
+  This means that:
+  - Even if your build only needs a few files,
+    changing _any_ file in the directory requires rebuilding the derivation,
+    potentially wasting a lot of time.
+
+  - If you have any secrets in the current directory,
+    they get imported into the Nix store too, exposing them to all users on the system!
+
+### `builtins.path`
```

Comment:
This section should be removed, agreed, it's not good practice to use `builtins.path` (or `lib.cleanSourceWith`). In the future we can at least potentially deprecate `cleanSourceWith`.